include/swift/SIL/SILBuilder.h

```diff
@@ -526,8 +575,11 @@ class SILBuilder {
 
   UncheckedRefCastInst *createUncheckedRefCast(SILLocation Loc, SILValue Op,
                                                SILType Ty) {
-    return insert(new (F.getModule()) UncheckedRefCastInst(
+    auto *I = insert(new (F.getModule()) UncheckedRefCastInst(
         getSILDebugLocation(Loc), Op, Ty));
+    I->getTypeDefOperandsBuf()[0].set(
+        getOpenedArchetypeDef(Ty, getModule(), OpenedArchetypes));
+    return I;
```

Comment:
Just because an instruction only uses one type doesn't mean that type is expressed in terms of <= 1 opened archetype.  You can absolutely cast to something like ConcreteClassType<@opened("foo") T, @opened("bar") U>.